pyOCD/gdbserver/gdbserver.py

```diff
@@ -363,11 +391,15 @@ def run(self):
                         if self.persist:
                             break
                         else:
-                            self._cleanup()
+                            self.shutdown_event.set()
                             return
 
                 self.lock.release()
 
+            except Exception as e:
```

Comment:
If an exception is thrown here then the current command won't be able to return a response of failure.  Would it make sense to respond with a generic error here?  Or maybe an exception could be handled at the individual command level.